runtime/lua/vim/_defaults.lua

```diff
@@ -155,6 +155,30 @@ do
     end,
   })
 
+  vim.api.nvim_create_autocmd('TermRequest', {
+    group = nvim_terminal_augroup,
+    desc = 'Check for an OSC foreground/background color request, and respond accordingly',
+    callback = function(args)
+      local fg_request = args.data == '\027]10;?'
+      local bg_request = args.data == '\027]11;?'
+      if fg_request or bg_request then
+        -- WARN: This does not return the actual foreground/background color,
+        -- but rather returns:
+        --   - fg=white/bg=black when Nvim option 'background' is 'dark'
+        --   - fg=black/bg=white when Nvim option 'background' is 'light'
+        local channel = vim.api.nvim_get_option_value('channel', { buf = args.buf })
+        local red, green, blue = 0, 0, 0
+        local bg_option_dark = vim.o.background == 'dark'
+        if (fg_request and bg_option_dark) or (bg_request and not bg_option_dark) then
+          red, green, blue = 65535, 65535, 65535
+        end
+        local command = fg_request and 10 or 11
+        local data = string.format('\027]%d;rgb:%04x/%04x/%04x\007', command, red, green, blue)
```

Comment:
Hrm, _technically_ we should  be using the same terminator that was used in the query (BEL or ST), but vterm strips the terminator by the time we get it, so there's no way to know what the original terminator was.

No action required here, just noting it for posterity.